### PR TITLE
[Login Performance]LPS-150112

### PR DESCRIPTION
**Pull request by tinatian**
This pull brings 3.1% percent improvement on Login

### CODE REVIEW

**Comment by tinatian** [comment]
FYI @BryanEngler

**Comment by tinatian** [comment]
@vicnate5 Please help me scan test results.

Thanks.
Tina.

**Comment by vicnate5** [comment]
@tinatian No regressions found.

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/shared/search/PortletSharedSearchRequestImpl.java [hunk 1]
```diff
@@ -253,33 +265,45 @@ private SearchSettingsContributor _getSearchSettingsContributor(
...
-		).filter(
-			Optional::isPresent
-		).map(
-			Optional::get
-		);
+		List<SearchSettingsContributor> searchSettingsContributors =
+			new ArrayList<>();
+
+		for (Portlet portlet : portlets) {
+			Optional<SearchSettingsContributor>
```
Comment (shuyangzhou) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/shared/search/PortletSharedSearchRequestImpl.java:278]: @slnn please get rid of this kind of Optional usage too. If the optional is at API level with many usages, you can leave them alone for now, But for all private usages, remove them all. You can just append commit to this pull.
Thanks!

Comment (slnn) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/shared/search/PortletSharedSearchRequestImpl.java:278]: OK, got it

Comment (slnn) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/shared/search/PortletSharedSearchRequestImpl.java:278]: Resend in https://github.com/liferay-core-infra/liferay-portal/pull/580

**Pull request closed without merging**